server/src/main/java/org/elasticsearch/gateway/PersistedClusterStateService.java

```diff
@@ -984,6 +1004,24 @@ void commit(long currentTerm, long lastAcceptedVersion, Version oldestIndexVersi
             ensureOpen();
             prepareCommit(currentTerm, lastAcceptedVersion, oldestIndexVersion);
             completeCommit();
+            assert assertOnCommit();
+        }
+
+        private boolean assertOnCommit() {
+            if (assertOnCommit != null /* TODO REVERT ME && Randomness.get().nextInt(100) == 0 */) {
```

Comment:
Actually I'm going to leave this in place until the backports are done and then reduce the frequency in a separate PR + backports, otherwise we might miss something and make the older branches flaky.